src/migration/1714633817175-SquadTotalPostsTrigger.ts

```diff
@@ -0,0 +1,65 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class SquadTotalPostsTrigger1714633817175 implements MigrationInterface {
+  name = 'SquadTotalPostsTrigger1714633817175';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION increment_squad_posts_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          UPDATE  source
+          SET     flags = jsonb_set(flags, '{totalPosts}', to_jsonb(COALESCE(CAST(flags->>'totalPosts' AS INTEGER) , 0) + 1))
+          WHERE   id = NEW."sourceId"
+          AND     type = 'squad';
+          RETURN NEW;
+        END;
+        $$
+      `);
+    queryRunner.query(
+      `CREATE TRIGGER increment_squad_posts_count AFTER INSERT ON "post" FOR EACH ROW EXECUTE PROCEDURE increment_squad_posts_count()`,
+    );
+
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION deduct_squad_stats()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        BEGIN
+          IF OLD.deleted = false AND NEW.deleted = true THEN
+            UPDATE  source
+            SET     flags = jsonb_set(
+                              jsonb_set(
```

Comment:
Why two `jsonb_set`? Should it not the second one be separate trigger? That is how the rest were done.

Reply:
This is to set two properties of the same flags property.

Reply:
Yeah sure, I just think move it to separate trigger I think so only field is updated in each trigger, as you did for others.

Reply:
Also @sshanzel what happens if post gets un-deleted. I am fine in not covering that case because it was not mentioned in RFC as well but we have to make a note of it somewhere.

Reply:
We are not doing any deduction on the stats when a post gets deleted. Should make our lives easier on that edge case.